OSDOCS-7670: adds 4.13.12 to RNs (#64545)
openshift/openshift-docs: release_notes/ocp-4-13-release-notes.adoc

==== Updating

To update an existing {product-title} 4.13 cluster to this latest release, see xref:../updating/updating-cluster-cli.adoc#updating-cluster-cli[Updating a cluster using the CLI] for instructions.

[id="ocp-4-13-12"]
=== RHBA-2023:5011 - {product-title} 4.13.12 bug fix

Issued: 2023-09-12

> {product-title} release 4.13.12 is now available. The list of bug fixes that are included in the update is documented in the link:https://access.redhat.com/errata/RHBA-2023:5011[RHBA-2023:5011] advisory. The RPM packages that are included in the update are provided by the link:https://access.redhat.com/errata/RHBA-2023:5014[RHBA-2023:5014] advisory.

Reviewer: FYI, `https://access.redhat.com/errata/RHBA-2023:5011` and `https://access.redhat.com/errata/RHBA-2023:5014` open a page not found.

Author: No worries, the release hasn't shipped yet. They will work once things ship. Thanks for the review.

> Space precluded documenting all of the container images for this release in the advisory.

Reviewer: Should line 3166 be commented out?

Author: No that is our standard text.

You can view the container images in this release by running the following command:

[source,terminal]
----
$ oc adm release info 4.13.12 --pullspecs
----

[id="ocp-4-13-12-features"]
==== Features

[id="ocp-4-13-12-networking-sriov-exclude-topology"]
===== Exclude SR-IOV network topology for NUMA-aware scheduling

With this release, you can exclude advertising the Non-Uniform Memory Access (NUMA) node for the SR-IOV network to the Topology Manager. By not advertising the NUMA node for the SR-IOV network, you can permit more flexible SR-IOV network deployments during NUMA-aware pod scheduling.

For example, in some scenarios, it is a priority to maximize CPU and memory resources for a pod on a single NUMA node. By not providing a hint to the Topology Manager about the NUMA node for the pod’s SR-IOV network resource, the Topology Manager can deploy the SR-IOV network resource and the pod CPU and memory resources to different NUMA nodes. In earlier {product-title} releases, the Topology Manager attempted to place all resources on the same NUMA node only.

> For more information about this more flexible SR-IOV network deployment during NUMA-aware pod scheduling, see xref:../networking/hardware_networks/configuring-sriov-device.adoc#nw-sriov-exclude-topology-manager_configuring-sriov-device[Exclude the SR-IOV network topology for NUMA-aware scheduling].

Reviewer: FYI, the page that this xref points to doesn't mention NUMA-aware pod scheduling. Is it the right page?

[id="ocp-4-13-12-custom-rhcos-image-for-gcp"]
===== Using a custom {op-system-first} image for a Google Cloud Provider cluster

> By default, the installation program downloads and installs the {op-system-first} image that is used to boot control plane and compute machines. With this enhancement, you can now override the default behavior by modifying the installation configuration file (install-config.yaml) to specify a custom {op-systm} image. Before you deploy the cluster, you can modify the following installation parameters:

Reviewer: ```suggestion
By default, the installation program downloads and installs the {op-system-first} image that is used to start control plane and compute machines. With this enhancement, you can now override the default behavior by modifying the installation configuration file (`install-config.yaml`) to specify a custom {op-system} image. Before you deploy the cluster, you can modify the following installation parameters:
```
IBM Style lists "boot" as a use-with-caution term and provides this guidance: "Although _boot_ is widely recognized among technical users, where usage permits use _start_ or _turn on_."

Should `{ocp-systm}` be `{ocp-system}`?

* `controlPlane.platform.gcp.osImage.project`
* `controlPlane.platform.gcp.osImage.name`
* `compute.platform.gcp.osImage.project`
* `compute.platform.gcp.osImage.name`
* `platform.gcp.defaultMachinePlatform.osImage.project`
* `platform.gcp.defaultMachinePlatform.osImage.name`

> For more information about these parameters, xref:../installing/installing_gcp/installing-gcp-customizations.adoc#installation-configuration-parameters-additional-gcp_installing-gcp-customizations[Additional Google Cloud Platform configuration parameters].

Reviewer: ```suggestion
For more information about these parameters, see xref:../installing/installing_gcp/installing-gcp-customizations.adoc#installation-configuration-parameters-additional-gcp_installing-gcp-customizations[Additional Google Cloud Platform configuration parameters].
```

[id="ocp-4-13-12-allocatedloabalancernodepors-network-api"]
===== Support for allocateLoadBalancerNodePorts in Service object of Network API

> The `ServiceSpec` component in Network API under `Service` object describes the attributes that a user creates on a service. The `allocateLoadBalancerNodePorts` attribute within `ServiceSpec` is now supported in {product-title} 4.13 release. The `allocateLoadBalancerNodePorts` attribute defines whether the `NodePorts` will be automatically allocated for services with type `LoadBalancer`.

Reviewer: ```suggestion
The `ServiceSpec` component in Network API under the `Service` object describes the attributes that a user creates on a service. The `allocateLoadBalancerNodePorts` attribute within the `ServiceSpec` component is now supported in {product-title} 4.13. The `allocateLoadBalancerNodePorts` attribute defines whether the `NodePorts` will be automatically allocated for services of the `LoadBalancer` type.
```

[id="ocp-4-13-12-bug-fixes"]
==== Bug fixes

> * Previously, the {product-title} router directed traffic to a route with a weight of 0 when it had only one backend. With this update, the router will not send traffic to routes with a single backend with weight 0. (link:https://issues.redhat.com/browse/OCPBUGS-17107[*OCPBUGS-17107*])

Reviewer: ```suggestion
* Previously, the {product-title} router directed traffic to a route with a weight of 0 when it had only one back end. With this update, the router will not send traffic to routes with a single back end with weight 0. (link:https://issues.redhat.com/browse/OCPBUGS-17107[*OCPBUGS-17107*])
```
If "backend" (one word, unhyphenated) is the accepted and common spelling in OCP, feel free to use that spelling. IBM Style lists "backend" as a use-with-caution term and provides this guidance: "In most cases, write as shown: two words. If possible, use a more specific term, such as _server_, _operating system_, or _network_. Where _backend_ is part of the established product terminology, for example in some compilers or in the Python programming language, write as one word with no hyphen."
